cmd/helm/repo_add.go

```diff
@@ -131,6 +135,23 @@ func addRepository(name, url, username, password string, home helmpath.Home, cer
 		return fmt.Errorf("Looks like %q is not a valid chart repository or cannot be reached: %s", url, err.Error())
 	}
 
+	// Lock the repository file for concurrent processes synchronization and re-read its content before updating it
+	fileLock := flock.New(home.RepositoryFile())
+	lockCtx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
```

Comment:
I'm wondering if `5*time.Minute` is too long to wait. I've never seen the operation take more than a few seconds. Is there a reason for such a long time?

Reply:
I've never seen it take so much time either. It shouldn't happen unless another process is holding a lock on the file for that long. Or the file had grown in size ludicrously.

I only choose 5 minutes to make sure it is not aborted too early.
Maybe there could be a parameter to allow changing this from the CLI with a way lower default that could be raised in case of issues.

Reply:
I've lowered it to 30 seconds for now.